tests/listener-removed.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const EventListenerTest = require('@janiscommerce/event-listener-test');
+const { ServerlessHandler } = require('@janiscommerce/event-listener');
+const Model = require('@janiscommerce/model');
+const mockRequire = require('mock-require');
+const path = require('path');
+
+const { ListenerRemoved, ModelClient } = require('../lib');
+
+const fakeClientPath = path.join(process.cwd(), process.env.MS_PATH || '', 'models', 'client');
+
+const handler = (...args) => ServerlessHandler.handle(ListenerRemoved, ...args);
+
+describe('Client Removed Listener', async () => {
+
+	const client = {
+		code: 'test1',
+		databases: { default: [Object] },
+		dateCreated: '2020-11-27T12:40:28.917Z',
+		dateModified: '2020-11-27T19:23:25.624Z',
+		status: 'active',
+		id: '5fc0f3bc617a1b3e98009c4c'
+	};
+
+	const validEvent = {
+		client: client.code,
+		service: 'id',
+		entity: 'client',
+		event: 'removed'
+	};
+
+	const sandboxAssertGetBy = sandbox => {
+		sandbox.assert.calledOnceWithExactly(ModelClient.prototype.getBy, 'code', client.code, { unique: true });
+	};
+
+
+	await EventListenerTest(handler, [
+		{
+			description: 'Should return 400 when the event has no client',
+			event: {
+				...validEvent,
+				client: undefined
+			},
+			responseCode: 400
+		},
+		{
+			description: 'Should return 500 when client model fails getting the client',
+			event: validEvent,
+			before: sandbox => {
+
+				mockRequire(fakeClientPath, ModelClient);
+
+				sandbox.stub(ModelClient.prototype, 'getBy').rejects();
+
+				sandbox.spy(Model.prototype, 'dropDatabase');
+
+				sandbox.spy(ModelClient.prototype, 'remove');
+
+				sandbox.spy(ListenerRemoved.prototype, 'postRemovedHook');
+			},
+			after: sandbox => {
+
+				sandboxAssertGetBy(sandbox);
+				sandbox.assert.notCalled(Model.prototype.dropDatabase);
+				sandbox.assert.notCalled(ModelClient.prototype.remove);
+				sandbox.assert.notCalled(ListenerRemoved.prototype.postRemovedHook);
+				mockRequire.stop(fakeClientPath);
+			},
+			responseCode: 500
+		},
+		{
+			description: 'Should return 500 when model default client fails to dropping the database',
```

Comment:
En inglés sería: _Should return 500 when it default model fails to **drop** de database_ Lo podes cambiar en los ejemplos de más abajo :+1: